app/src/main/java/ca/wheresthebus/ui/home/HomeFragment.kt

```diff
@@ -162,13 +162,16 @@ class HomeFragment : Fragment() {
     private fun refreshBusTimes() {
         lifecycleScope.launch(Dispatchers.IO) {
             try {
-                // Create a map of favourite stops to their next bus times
+                // Convert all favorite stops to pairs list and do GTFS realtime call
+                val stopRoutePairs = favouriteStopsList.map { stop ->
+                    StopId(stop.busStop.id.value) to RouteId(stop.route.id.value)
+                }
+                val busTimesLists = GtfsRealtimeHelper.getBusTimes(stopRoutePairs)
+
+                // Create a mutableMap of stop codes to their next bus times
                 val busTimesMap = mutableMapOf<StopCode, List<Duration>>()
-                favouriteStopsList.forEach { stop ->
-                    val nextBusTimes = GtfsRealtimeHelper.getBusTimes(
-                        StopId(stop.busStop.id.value),
-                        RouteId(stop.route.id.value))
-                    busTimesMap[stop.busStop.code] = nextBusTimes
+                for (i in busTimesLists.indices) {
+                    busTimesMap[favouriteStopsList[i].busStop.code] = busTimesLists[i]
```

Comment:
Can getBusTimes instead return a map of stop codes to durations? I don't see when returning a list of lists would be useful.

Reply:
Do you mean return a map of stop codes to a list of durations? The list of lists is under the assumption that the index matches the pairs of the objects that are passed in.